build_helpers/publishPackage.sh

```diff
@@ -1,35 +1,70 @@
 #!/bin/bash
+# example usages:
+# ./publishPackage.sh --branch=v1.0.x
+# ./publishPackage.sh --branch=v1.1.x --latest
+# ./publishPackage.sh --branch=v1.0-beta --beta
+
 set -e
 
 # Defaults that can be overridden in the CLI (except the cookie file)
-export BETA=0
+export BETA=false
+export LATEST=false
+export BRANCH=""
 
+# Parse flags
 while (( "$#" )); do
   opt=$1
   VALUE=$2
   # Make sure to shift here and any option that uses the $VALUE
   shift
   case $opt in
-    --beta) BETA=$VALUE; shift ;;
+    --beta)
+      BETA=true
+      ;;
+
+    --latest)
+      LATEST=true
+      ;;
+
+    --branch)
+      if [ -z $VALUE ] ; then
+        echo "Please specify branch name"
+        exit 192
+      fi
+      BRANCH=$VALUE;
+      shift
+      ;;
+
     * ) # Invalid flag
       echo "Invalid flag: $opt"
       exit 192
       ;;
   esac
 done
 
-current_version=$(node -p "require('./package').version")
+# branch name is required
+if [ -z "$BRANCH" ] ; then
+  echo "Please specify branch name using the --branch flag"
+  exit 192
+fi
 
-printf "Next version (current is $current_version)? "
-read next_version
+current_version=$(node -p "require('./package').version")
 
-if ! [[ $next_version =~ ^[0-9]\.[0-9]+\.[0-9](-.+)? ]]; then
-  echo "Version must be a valid semver string, e.g. 1.0.2 or 2.3.0-beta.1"
-  exit 1
-fi
+# prompt next version
+while [ true ]; do
```

Comment:
Re-prompt if user enters invalid version.

Reply:
I think the current behaviour is better.  May be we can at max provide couple of more attempts in case of incorrect input but not an infinite retry.